### PR TITLE
[test_system_health] Align system health test cases with production code and fix issues

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
### Description of PR

Summary:

1. Aligh system health test cases with latest production code
2. Fix issue when mocking fan speed value
3. Combine multiple mock together to save test time

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?

Fix test cases to align with latest production code

#### How did you do it?

1. Aligh system health test cases with latest production code
2. Fix issue when mocking fan speed value
3. Combine multiple mock together to save test time

#### How did you verify/test it?

Manually run the test cases

#### Any platform specific information?

N/A

#### Supported testbed topology if it's a new test case?

N/A

### CODE REVIEW

File: tests/system_health/test_system_health.py [hunk 1]
```diff
@@ -100,129 +101,108 @@ def test_device_checker(duthosts, rand_one_dut_hostname, device_mocker_factory):
...
+        psu_mock_result, psu_name = device_mocker.mock_psu_presence(False)
+        psu_expect_value = EXPECT_PSU_MISSING.format(psu_name)
 
-        mock_result, fan_name = device_mocker.mock_fan_status(False)
-        expect_value = EXPECT_FAN_BROKEN.format(fan_name)
-        if mock_result:
-            logging.info('Mocked fan broken for {}, waiting {} seconds for it to take effect'.format(fan_name,
-                                                                                                     THERMAL_CHECK_INTERVAL))
+        if fan_mock_result and asic_mock_result and psu_mock_result:
+            logging.info('Mocked invalid fan speed for {}'.format(fan_name))
```
Comment (wangxin) [inline tests/system_health/test_system_health.py:114]: Can you update the code to create a module specific logger instead of using the root logger?

Comment (Junchao-Mellanox) [reply tests/system_health/test_system_health.py:114]: Sure

Comment (Junchao-Mellanox) [reply tests/system_health/test_system_health.py]: Done

**Review by wangxin — APPROVED** [review]

**Pull request merged**